### ISSUE TITLE
Vetur dosen't support import component folder automatically?

### ISSUE THREAD

**Issue by abitwhy** [vuejs/vetur#3247]
Hello, dear Vuter maintainer.
As the title says, Vetur cannot automatically import the component, which is a folder, such as src/components/Foo/index.vue. Is Vetur not supporting it? or need additional configuration? If it is not supported, Could you consider providing support? Because a component is composed of multiple internal components should be regarded as a relatively common requirement.

**Comment by yoyo930021** [comment]
https://vuejs.github.io/vetur/guide/global-components.html#example

**Comment by abitwhy** [comment]
I have noticed this method, but it needs to be configured for each component, which is not the expected automatic import.

**Comment by yoyo930021** [comment]
> I have noticed this method, but it needs to be configured for each component, which is not the expected automatic import.

Please using `'./src/components/**/*.vue'`.
Or Using `require('fs')` to get all components in config.

It is a custom config for the component name format.

**Comment by abitwhy** [comment]
There is no automatic code completion, which is my motivation for raising an issue. Thank you for your accurate and timely answer. By the way, I understand [the globalComponents configuration](https://vuejs.github.io/vetur/guide/global-components.html#example).

**Comment by yoyo930021** [comment]
> There is no automatic code completion, which is my motivation for raising an issue. Thank you for your accurate and timely answer. By the way, I understand [the globalComponents configuration](https://vuejs.github.io/vetur/guide/global-components.html#example).

Vetur can't know how are you register global components.
Everyone has their own favorite way of setting up.
Vetur only provides a flexible way for you to set.

**Comment by abitwhy** [comment]
Let me rephrase the problem.
### folder
```
├─components
│ ├─Foo-------------------------- // component folder
│ │ ├─index.vue
│ │ ├─FooBar.vue
```
### usage
`<foo/>`

How to use the following features?

![Auto Import](https://user-images.githubusercontent.com/68452982/143201560-48f12a7c-3385-4930-85d4-e32a619a044b.png)

**Comment by abitwhy** [comment]
Use [the globalComponents configuration](https://vuejs.github.io/vetur/guide/global-components.html#example), or change the folder:

```
├─components
│ ├─Foo-------------------------- // component folder
│ │ ├─Foo.vue
│ │ ├─FooBar.vue
```
